integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -123,8 +123,16 @@ function describeRoundtrip({
     it("reads", async function (this: RealmContext) {
       await setupTest(this.realm);
 
-      const obj = this.realm.objectForPrimaryKey<MixedClass>("MixedClass", this._id);
-      if (!obj) throw new Error("Object not found");
+      const obj = await new Promise<MixedClass>((resolve) => {
+        this.realm
+          .objects<MixedClass>("MixedClass")
+          .filtered("_id = $0", this._id)
+          .addListener(([obj]) => {
+            if (obj) {
+              resolve(obj);
```

Comment:
Are we fine with not removing the listener?

Reply:
Sure. When the Realm gets closed after the test (the `openRealmBefore` hook does that) the listeners wont fire again. And even if they did, they could only resolve a promise that already got resolved, which is a no-op.